hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ReadOnlyZKClient.java

```diff
@@ -258,6 +264,23 @@ public void closed(IOException e) {
     }
   }
 
+  private static TimerTask getTimerTask(final long timeoutMs, final CompletableFuture<?> future,
+    final String api) {
+    return timeout -> {
+      if (!future.isDone()) {
+        future.completeExceptionally(new DoNotRetryIOException(
+          "Zookeeper " + api + " could not be completed in " + timeoutMs + " ms"));
+      }
+    };
+  }
+
+  public CompletableFuture<byte[]> getWithTimeout(final String path, final long timeoutMs) {
```

Comment:
nit: Can we overload the existing methods instead of creating new methods with Timeout suffix?
eg.
` public CompletableFuture<byte[]> get(final String path, final long timeoutMs) `